zerver/lib/events.py

```diff
@@ -528,14 +528,23 @@ def apply_event(state: Dict[str, Any],
                 state['plan_includes_wide_organization_logo'] = event['value'] != Realm.LIMITED
                 state['realm_upload_quota'] = event['extra_data']['upload_quota']
 
-            # Tricky interaction: Whether we can create streams can get changed here.
-            if (field in ['realm_create_stream_policy',
-                          'realm_waiting_period_threshold']) and 'can_create_streams' in state:
-                state['can_create_streams'] = user_profile.can_create_streams()
+            permission_types = ['can_create_streams', 'can_subscribe_other_users']
+            policy_types = ['create_stream_policy', 'invite_to_stream_policy']
+            policy_permission_dict = {policy_types[i]: permission_types[i] for i in range(len(policy_types))}
```

Comment:
This code is weird and hard to read -- why not just make create a hardcoded dictionary rather than doing this weird zipping of two lists to make it?